### ISSUE TITLE
Add ability to configure schemes

### ISSUE THREAD

**Issue by jcampbell05** [workshop/struct#63]
From https://github.com/jcampbell05/xcake/issues/124

Some features of schemes are useful for example flags for things such as Coredata debugging.

**Comment by lyptt** [comment]
Proposed spec adjustment as follows:

```yaml
schemes:
  My scheme:
    analyze:
      configuration:
        IPHONEOS_DEPLOYMENT_TARGET: 10.0
    archive:
      name: MyApp.xcarchive
      reveal: false
    build:
      parallel: true
      build_implicit: true
      targets:
        My app:
          archiving_enabled: true
          running_enabled: true
          profiling_enabled: true
          testing_enabled: true
    launch:
      simulate_location: true
      target: My App
      arguments: -AppleLanguages (en-GB)
      environment: 
        OS_ACTIVITY_MODE: disable
    profile:
      target: My app
      configuration:
        IPHONEOS_DEPLOYMENT_TARGET: 10.0
      inherit_environment: true
```

**Comment by lyptt** [comment]
Proposed Specfile adjustment as follows:

```ruby
scheme('My scheme') do
  analyze do
    configuration do
      override 'IPHONEOS_DEPLOYMENT_TARGET', '10.0'
    end
  end

  archive name: 'MyApp.xcarchive', reveal: false

  build do
    parallel: true
    build_implicit: true
    target('My app') do
      enable_archiving
      enable_running
      enable_profiling
      enable_testing
    end

  launch('My app') do
    enable_location_simulation
    arguments '-AppleLanguages (en-GB)'
    environment do
      override 'OS_ACTIVITY_MODE', '10.0'
    end
  end

  profile('My app') do
    configuration do
      override 'IPHONEOS_DEPLOYMENT_TARGET', '10.0'
    end
    inherit_environment
  end
end
```

**Comment by lyptt** [comment]
Fixed in 2.0.0
